### ISSUE TITLE
TypeError: IntlMessageFormat.__parse must be set 

### ISSUE THREAD

**Issue by Lleios** [formatjs/react-intl#1400]
### Expected behavior

No error with FormattedMessage

### Current behavior

All my "FormattedMessage" components throw this error even if they format text correcty : 

> index.js:1437 [React Intl] Error formatting message: "TEST.TEST" for locale: "en"
TypeError: IntlMessageFormat.__parse must be set to process `message` of type `string`

### Step to reproduce for BUG REPORT

```
import * as React from "react";
import { render } from "react-dom";
import { IntlProvider, FormattedMessage } from "react-intl";

const en = {
  TEST: "test"
};

function App() {
  return <FormattedMessage id="TEST" />;
}

render(
  <IntlProvider locale="en" messages={en}>
    <App />
  </IntlProvider>,
  document.getElementById("root")
);
```

### Environment

|          Executable | Version |
| ------------------: | :------ |
| `react-intl` | 3.1.1 |
|      `react` | 16.8.6 |
|     `npm` | 6.9.0 |
|    `node ` | 10.16.0 |
|    `typescript ` | 3.5.3 |

| OS   | Version |
| ---- | ------- |
| macOS Mojave | 10.14.6 |

| Browser | Version |
| ------- | ------- |
| Chrome    | 76.0.3809.100  |

**Comment by longlho** [comment]
I can't seem to repro it https://codesandbox.io/s/fervent-hypatia-f855r

**Comment by Lleios** [comment]
i tried on my mac, a colleague mac and a colleague pc on windows and they have the same error.
I don't know why on codesandbox it's work fine.
And i was wrong, text is not formatted correcty if i use "values"

this code render "test {name}"

```
import * as React from "react";
import { render } from "react-dom";
import { IntlProvider, FormattedMessage } from "react-intl";

const en = {
  TEST: "test {name}"
};

function App() {
  return (
    <>
      <FormattedMessage id="TEST" values={{ name: "test" }} />
    </>
  );
}

render(
  <IntlProvider locale="en" messages={en}>
    <App />
  </IntlProvider>,
  document.getElementById("root")
);
```

How IntlMessageFormat.parse could not be set?

**Comment by longlho** [comment]
What's the build toolchain you're using? Might be a tree-shaking thing where our parser got shaken off?

**Comment by Lleios** [comment]
i use https://github.com/wmonk/create-react-app-typescript 

I have tested all beta version and the latest working version is 3.0.0-beta.14. 

Does that confirm what you think?

**Comment by longlho** [comment]
hmm can u try upgrading `intl-messageformat`? we just released a new version

**Comment by s524797336** [comment]
```
├─┬ @elastic/eslint-plugin-react-intl@1.2.0
│ └── intl-messageformat-parser@1.8.1 
├─┬ babel-plugin-react-intl@4.1.8
│ └── intl-messageformat-parser@3.0.4 
├─┬ babel-plugin-react-intl-extractor@1.0.8
│ └─┬ babel-plugin-react-intl@3.5.1
│   └── intl-messageformat-parser@1.8.1  deduped
└─┬ UNMET PEER DEPENDENCY react-intl@3.1.2
  ├─┬ intl-messageformat@6.1.0
  │ └── intl-messageformat-parser@3.0.4 
  └── intl-messageformat-parser@3.0.4
```
Same, maybe the version dependency is wrrong with those plugin?

**Comment by Lleios** [comment]
It does'nt work for me

react-intl: 3.1.2
intl-messageformat : 6.1.0

**Comment by longlho** [comment]
Yeah looks like you have 2 babel-plugin-react-intl

**Comment by longlho** [comment]
Something's funky with your dependency chain, there're a bunch of older versions of things lying around...

**Comment by longlho** [comment]
feel free to reopen if u're still having this issue.

**Comment by Lleios** [comment]
it's fix with 3.1.6 :)
